contacts/decorators.py

```diff
@@ -0,0 +1,14 @@
+from functools import wraps
+from django.core.exceptions import PermissionDenied
+
+from .models import Contact
+
+
+def user_added_contact(view_func):
+    def _decorator(request, *args, **kwargs):
+        if Contact.objects.filter(pk=kwargs['pk'], user=request.user).exists():
```

Comment:
we should use `kwargs.get('pk')` so that it doesn't crash if we accidentally add it over a view not receiving `pk` from url.